app/controllers/search_controller.rb

```diff
@@ -4,13 +4,13 @@ class SearchController < SessionsController
   require 'will_paginate/array'
 
   def explore
-    @repositories = Repository.paginate(:per_page=>12,:page=>params[:page]).order([sort_order].to_h).page params[:page]
+    @repositories = Repository.paginate(:per_page=>12,:page=>params[:page]).where(share_type: "public").order([sort_order].to_h).page params[:page]
```

Comment:
This `where(share_type: "public)` should be a scope on the model, although I think it should be the default scope, which I think you can do.